users/vieting/experiments/switchboard/ctc/feat/experiments.py

```diff
@@ -401,6 +616,203 @@ def process_args(args: Dict[str, Any]):
+        """
+
+        key_string = ""
+        report_dict = {}
+
+        for key, value in args.items():
+
+            if key in ["speed", "tempo", "preemphasis", "non_linearity"]:
+                key_component = f"{key}_{value['prob']}_{value['minimum']}_{value['maximum']}"
+                if "default" in value:
+                    key_component += f"_{value['default']}"
+                key_string += key_component
+                report_dict[key] = f"{value['prob']}_{value['minimum']}_{value['maximum']}"
+            elif key == "codecs":
+                codecs_str = "_".join(
+                    [f"{codec['encoding']}_{codec['prob']}_{codec['minimum']}_{codec['maximum']}" for codec in value]
+                )
+                key_string += f"{key}_{codecs_str}_"
+                report_dict[key] = codecs_str
+            else:
+                raise ValueError(f"Unknown argument name: {key}")
+
+        return key_string, report_dict
+
+    nn_base_args = {}
+
+    for args in perturbation_args:
+        exp_name_suffix, report_dict = process_args(args)
+
+        # Construct the exp_name and report_args
+        exp_name = f"scf_bs2x5k_perturb_from_checkpoint_24_{exp_name_suffix}"
+        report_args = report_dict
+        nn_base_args[exp_name] = dict(
+            returnn_args={
+                "extra_args": {
+                    "audio_perturb_args": args,
+                    "audio_perturb_runner": CodeWrapper("WaveformPerturbation(**audio_perturb_args)"),
+                    "conv_pad_seq_len_to_power": 1.5,
+                    "accum_grad_multiple_step": 2,
+                    "preload_from_files": {
+                        "existing-model": {
+                            "filename": nn_system.train_jobs[
+                                "conformer_bs2x5k_scf_baseline_preemphasis97_wn"
+                            ].out_checkpoints[24],
+                            "init_for_train": True,
+                        }
+                    },
+                },
+                **returnn_args,
+            },
+            feature_args={
+                "class": "ScfNetwork",
+                "size_tf": 256 // 2,
+                "stride_tf": 10 // 2,
+                "preemphasis": 0.97,
+                "wave_norm": True,
+            },
+            lr_args=lr_args,
+            report_args=report_args,
+        )
+    for args in perturbation_args_preemphasis:
+        exp_name_suffix, report_dict = process_args(args)
+
+        # Construct the exp_name and report_args
+        exp_name = f"scf_bs2x5k_perturb_from_checkpoint_24_{exp_name_suffix}"
+        report_args = report_dict
+        nn_base_args[exp_name] = dict(
+            returnn_args={
+                "extra_args": {
+                    "audio_perturb_args": args,
+                    "audio_perturb_runner": CodeWrapper("WaveformPerturbation(**audio_perturb_args)"),
+                    "conv_pad_seq_len_to_power": 1.5,
```

Comment:
Is the gradient accumulation missing on purpose?

Reply:
No